storm-core/src/jvm/backtype/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -0,0 +1,296 @@
+    validateKey(key);
+    checkForBlobOrDownload(key);
+    SettableBlobMeta meta = getStoredBlobMeta(key);
+    _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
+    try {
+      fbs.deleteKey(DATA_PREFIX+key);
+      fbs.deleteKey(META_PREFIX+key);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public InputStreamWithMeta getBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException {
+    validateKey(key);
+    if(!checkForBlobOrDownload(key)) {
+      checkForBlobUpdate(key);
+    }
+    SettableBlobMeta meta = getStoredBlobMeta(key);
+    _aclHandler.hasPermissions(meta.get_acl(), READ, who, key);
+    try {
+      return new BlobStoreFileInputStream(fbs.read(DATA_PREFIX+key));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Iterator<String> listKeys() {
+    try {
+      return new KeyTranslationIterator(fbs.listKeys(), DATA_PREFIX);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void shutdown() {
+  }
+
+  @Override
+  public int getBlobReplication(String key, Subject who) throws Exception {
+    validateKey(key);
+    SettableBlobMeta meta = getStoredBlobMeta(key);
+    _aclHandler.hasPermissions(meta.get_acl(), READ, who, key);
+    CuratorFramework zkClient = BlobStoreUtils.createZKClient(conf);
+    if (zkClient.checkExists().forPath(BLOBSTORE_SUBTREE + key) == null) {
+       zkClient.close();
+       return 0;
+    }
+    int replicationCount = zkClient.getChildren().forPath(BLOBSTORE_SUBTREE + key).size();
+    zkClient.close();
+    return replicationCount;
+  }
+
+  @Override
+  public int updateBlobReplication(String key, int replication, Subject who) throws AuthorizationException, KeyNotFoundException {
+      throw new UnsupportedOperationException("For local file system blob store the update blobs function does not work. " +
+              "Please use HDFS blob store to make this feature available.");
+  }
+
+  //This additional check and download is for nimbus high availability in case you have more than one nimbus
+  public boolean checkForBlobOrDownload(String key) {
+    boolean checkBlobDownload = false;
+    try {
+      List<String> keyList = BlobStoreUtils.getKeyListFromBlobStore(this);
+      if (!keyList.contains(key)) {
+        CuratorFramework zkClient = BlobStoreUtils.createZKClient(conf);
+        if (zkClient.checkExists().forPath(BLOBSTORE_SUBTREE + key) != null) {
+          Set<NimbusInfo> nimbusSet = BlobStoreUtils.getNimbodesWithLatestSequenceNumberOfBlob(zkClient, key);
+          if (BlobStoreUtils.downloadMissingBlob(conf, this, key, nimbusSet)) {
+            LOG.debug("Updating blobs state");
+              BlobStoreUtils.createStateInZookeeper(conf, key, nimbusInfo);
+            checkBlobDownload = true;
+          }
+        }
+        zkClient.close();
```

Comment:
same as above, place `zkClient.close` in finally bock, same for all below if any.